app/services/crypto_payment_service.py

```
@@ -0,0 +1,558 @@
+                
+                if user_id and payment_id and credit_amount:
+                    # Process credit allocation
+                    await self._allocate_credits_to_user(user_id, int(credit_amount), payment_id)
+                    
+                    return {
+                        "success": True,
+                        "user_id": user_id,
+                        "credits_allocated": credit_amount
+                    }
+            
+            return {"success": True, "message": "Webhook processed but no action needed"}
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)